libs/datamodel/core/src/transform/ast_to_dml/db/walkers.rs

```diff
@@ -244,7 +271,16 @@ impl<'ast, 'db> RelationFieldWalker<'ast, 'db> {
                 db: self.db,
                 scalar_field: &self.db.types.scalar_fields[&(model_id, *field_id)],
             }
-        })
+        };
+
+        match self.relation_field.references.as_ref() {
+            Some(references) => references.iter().map(f),
+            None => [].iter().map(f),
+        }
+    }
+
+    pub(crate) fn is_compound_relation(&self) -> bool {
+        self.referenced_fields().len() > 1
```

Comment:
This is the sort of methods I am worrying about, because that shouldn't land in a public API (doesn't take into account relations where `references` is left implicit)

Reply:
Luckily it's not a public api yet :)